node/db/src/rocks.rs

```diff
@@ -40,19 +40,43 @@ impl RocksDb {
         if let Some(max_background_jobs) = config.max_background_jobs {
             db_opts.set_max_background_jobs(max_background_jobs);
         }
-        if let Some(compaction_style) = &config.compaction_style {
-            db_opts.set_compaction_style(compaction_style_from_str(compaction_style).unwrap());
+        if let Some(compaction_style) = compaction_style_from_str(&config.compaction_style).unwrap()
+        {
+            db_opts.set_compaction_style(compaction_style);
             db_opts.set_disable_auto_compactions(false);
         } else {
             db_opts.set_disable_auto_compactions(true);
         }
-        if let Some(compression_type) = &config.compression_type {
-            db_opts.set_compression_type(compression_type_from_str(compression_type).unwrap());
-        }
+        db_opts.set_compression_type(compression_type_from_str(&config.compression_type).unwrap());
         if config.enable_statistics {
+            db_opts.set_stats_dump_period_sec(config.stats_dump_period_sec);
```

Comment:
maybe `stats_dump_period_sec` could be an `Option`, so we wouldn't need `enable_statistics`, as they convey the same logic. Not a big deal though.